p2p/network.go

```diff
@@ -0,0 +1,152 @@
+package p2p
```

Comment:
What's the idea behind having a different `p2p` package to the `network` package?